jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/EtagUtil.java

```diff
@@ -0,0 +1,194 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Objects;
+
+import org.eclipse.jetty.util.QuotedStringTokenizer;
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Utility classes for Etag behaviors
+ */
+public class EtagUtil
+{
+    /**
+     * <p>The separator within an etag used to indicate a compressed variant.</p>
+     *
+     * <p>Default separator is {@code --}</p>
+     *
+     * <p>Adding a suffix of {@code gzip} to an etag of {@code W/"28c772d6"} will result in {@code W/"28c772d6--gzip"}</p>
+     *
+     * <p>The separator may be changed by the
+     * {@code org.eclipse.jetty.http.EtagUtil.ETAG_SEPARATOR} System property. If changed, it should be changed to a string
+     * that will not be found in a normal etag or at least is very unlikely to be a substring of a normal etag.</p>
+     */
+    public static final String ETAG_SEPARATOR = System.getProperty(EtagUtil.class.getName() + ".ETAG_SEPARATOR", "--");
+
+    /**
+     * Calculate the weak etag for the provided {@link Path} reference.
+     *
+     * <p>
+     * This supports relative path references as well.
+     * Which can be useful to establish a reliable etag if the base changes.
+     * </p>
+     *
+     * @param path the path to calculate from
+     * @return the weak etag
+     */
+    public static String calcWeakEtag(Path path)
```

Comment:
Shouldn't this method accept a `Resource` instead of a `Path`?

Reply:
Different method supports `Resource` now, this method still exists.